### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: improves unit test coverage

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:  improves #2576

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [x] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/acsengine/defaults-apiserver_test.go [hunk 1]
```diff
@@ -352,71 +351,3 @@ func TestAPIServerConfigDefaultAdmissionControls(t *testing.T) {
 		t.Fatalf("Admission control key '%s' not set in API server config for version %s", enableAdmissionPluginsKey, version)
 	}
 }
-
-func createContainerService(containerServiceName string, orchestratorVersion string, masterCount int, agentCount int) *api.ContainerService {
```
Comment (CecileRobertMichon) [inline pkg/acsengine/defaults-apiserver_test.go]: refactored in common place, function was duplicated

File: pkg/acsengine/output.go [hunk 2]
```diff
@@ -114,6 +114,9 @@ func (w *ArtifactWriter) WriteTLSArtifacts(containerService *api.ContainerServic
 			return e
 		}
 		for i := 0; i < properties.MasterProfile.Count; i++ {
+			if len(properties.CertificateProfile.EtcdPeerPrivateKeys) <= i || len(properties.CertificateProfile.EtcdPeerCertificates) <= i {
```
Comment (CecileRobertMichon) [inline pkg/acsengine/output.go]: Guard against index out of bound

File: pkg/acsengine/types.go [hunk 3]
```diff
@@ -101,3 +103,86 @@ type KeyVaultRef struct {
 }
 
 type paramsMap map[string]interface{}
+
+// CreateMockContainerService returns a mock container service for testing purposes
+func CreateMockContainerService(containerServiceName, orchestratorVersion string, masterCount, agentCount int, certs bool) *api.ContainerService {
```
Comment (jackfrancis) [inline pkg/acsengine/types.go]: Let's move this func to another file, perhaps `mocks.go`?

Comment (CecileRobertMichon) [reply pkg/acsengine/types.go]: done

File: cmd/root.go [hunk 4]
```diff
@@ -45,24 +45,7 @@ func NewRootCmd() *cobra.Command {
 	rootCmd.AddCommand(newUpgradeCmd())
 	rootCmd.AddCommand(newScaleCmd())
 	rootCmd.AddCommand(newDcosUpgradeCmd())
-
-	var completionCmd = &cobra.Command{
```
Comment (CecileRobertMichon) [inline cmd/root.go]: moved this to a function, fyi @stuartleeks @jackfrancis

Comment (stuartleeks) [reply cmd/root.go]: Cool :-)

**Pull request merged**